packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -62,6 +90,46 @@ class ComponentStateBucket {
 }
 
 class CurlyComponentManager {
+  prepareArgs(definition, args) {
+    validatePositionalParameters(args.named, args.positional.values, definition.ComponentClass.positionalParams);
+
+    if (definition.args) {
+      // args (aka 'oldArgs') may be undefined or simply be empty args, so
+      // we need to fall back to an empty array or object.
+      let newNamed = args && args.named && args.named.map || {};
+      let newPositional = args && args.positional && args.positional.values || [];
```

Comment:
(I don't even think `args.named` and `args.positional` can be null, but the code seems to suggest that they could be)